File: budget.cpp
#include <iostream>
#include <string>
using namespace std;

struct monthBudget{
    //Spending
    float rent; //House
    float carInsurance; //Car
    float electricity; //House
    float otherUtilities; //House
    float wiFi; //House
    float groceries; // House
    float foodEntertainment; //Personal
    float funEntertainment; //Personal
    float gas; //Car
    float debt; //Personal
    float phone; //Personal
    float gym; //Personal
    float emma;
    float savings;
    float income;

    void setIncome(){
        cout << "How much money do you have this month? ";
            cin >> income;
        cout << "What percentage of that would you like to save? (ENTER AS DECIMAL) ";
            cin >> savings;
        cout << endl;
    }

    float setVehicleExpenses(){
        carInsurance = 120.00;

        float milesDrive = (5 * 60) + (2 * 50);
        float mpg = 33.0;
        float gallons = milesDrive / mpg;
        gas = 4 * (gallons * 3.00);

        float totalVehical = carInsurance + gas;
        return totalVehical;
    }

    float setHousingExenses(){ //ANOTHER IDEA: set everything but rent and calculate rent at the end relative to everything else
        electricity = 100.00; //100
        wiFi = 60.00; //60
        otherUtilities = 50.00; //50
        groceries = 500; //Split-ish? //400
        float totalHousing = electricity + wiFi + otherUtilities + groceries;
        return totalHousing;
    }

    float setPersonalExpenses(){
        foodEntertainment = 350; //350
        funEntertainment = 150; //150
        emma = 50;
        debt = 250; //250
        phone = 10.00; //10
        gym = 200; //200
        float totalPersonal = foodEntertainment + funEntertainment + debt + phone + gym + emma;
        return totalPersonal;
    }

    float calcRemainder (float veh, float hou, float per){
        float remainder = income * (1 - savings);
        remainder = remainder - veh - hou - per;
        return remainder;
    }

    void calcRent (float remainder){
        cout << "I can spend up to: " << remainder << " on rent\n"; 
    }
};

int main(){
    monthBudget month;
    month.setIncome();
    float vehicleExpenses = month.setVehicleExpenses();
    float housingExpenses = month.setHousingExenses();
    float personalExpenses = month.setPersonalExpenses();
    float remainder = month.calcRemainder(vehicleExpenses, housingExpenses, personalExpenses);
    month.calcRent(remainder);
    cout << "Saving: $" << (month.income * month.savings) << " per month \n\n";


}

File: budget.py
class MonthBudget:
    def __init__(self):
        # Spending
        self.rent = 0.0  # House
        self.carInsurance = 0.0  # Car
        self.electricity = 0.0  # House
        self.otherUtilities = 0.0  # House
        self.wiFi = 0.0  # House
        self.groceries = 0.0  # House
        self.foodEntertainment = 0.0  # Personal
        self.funEntertainment = 0.0  # Personal
        self.gas = 0.0  # Car
        self.debt = 0.0  # Personal
        self.phone = 0.0  # Personal
        self.gym = 0.0  # Personal
        self.emma = 0.0
        self.savings = 0.0
        self.income = 0.0

    def set_income(self):
        self.income = float(input("How much money do you have this month? "))
        self.savings = float(input("What percentage of that would you like to save? (ENTER AS DECIMAL) "))
        print()

    def set_vehicle_expenses(self):
        self.carInsurance = 120.00

        miles_drive = (5 * 60) + (2 * 50)
        mpg = 30.0
        gallons = miles_drive / mpg
        self.gas = 4 * (gallons * 3.00)

        total_vehicle = self.carInsurance + self.gas
        return total_vehicle

    def set_housing_expenses(self):
        self.electricity = 100.00  # 100
        self.wiFi = 60.00  # 60
        self.otherUtilities = 50.00  # 50
        self.groceries = 500.00  # Split-ish? # 400
        total_housing = self.electricity + self.wiFi + self.otherUtilities + self.groceries
        return total_housing

    def set_personal_expenses(self):
        self.foodEntertainment = 350.00  # 350
        self.funEntertainment = 150.00  # 150
        self.emma = 50.00
        self.debt = 250.00  # 250
        self.phone = 10.00  # 10
        self.gym = 200.00  # 200
        total_personal = self.foodEntertainment + self.funEntertainment + self.debt + self.phone + self.gym + self.emma
        return total_personal

    def calc_remainder(self, veh, hou, per):
        remainder = self.income * (1 - self.savings)
        remainder = remainder - veh - hou - per
        return remainder

    def calc_rent(self, remainder):
        print("I can spend up to:", remainder, "on rent\n")


def main():
    month = MonthBudget()
    month.set_income()
    vehicle_expenses = month.set_vehicle_expenses()
    housing_expenses = month.set_housing_expenses()
    personal_expenses = month.set_personal_expenses()
    remainder = month.calc_remainder(vehicle_expenses, housing_expenses, personal_expenses)
    month.calc_rent(remainder)
    print("Saving: $", month.income * month.savings, "per month\n")


if __name__ == "__main__":
    main()
